### PR TITLE
llvmPackages_rocm: 4.5.2 -> 5.0.2

### PR DESCRIPTION
**Pull request by acowley**
hip: 4.5.2 -> 5.0.2

###### Description of changes

llvmPackages_rocm (including `llvm`, `clang`, `compiler-rt`, and `lld`) upgraded from ROCm 4.5.2 to 5.0.2.

The `hip` compiler is also upgraded along the same path as it is coupled with `llvmPackages_rocm.clang`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by acowley** [comment]
Hi @InternetUnexplorer, if you have a chance, can you also give this a whirl? This comes in atop your recent change to switch to static linking, and I'd like to be sure this update doesn't set things back (it shouldn't!).

**Comment by InternetUnexplorer** [comment]
Sure, I'll give it a go as soon as I can (will probably be tomorrow morning) :)

**Review by InternetUnexplorer — APPROVED** [review]
It took a bit longer than expected, but I can confirm that there seems to be no regression (which, like you said, is expected, but it's still nice to make sure) :)

File: pkgs/development/compilers/llvm/rocm/default.nix [hunk 1]
```diff
@@ -47,7 +47,7 @@ in rec {
 
   compiler-rt = callPackage ./compiler-rt {
     inherit version llvm;
-    src = "${src}/compiler-rt";
+    src = "${src}";
```
Comment (SuperSandro2000) [inline pkgs/development/compilers/llvm/rocm/default.nix:50]:
```suggestion
    inherit src;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 165362` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
14 packages built:
</summary>
<ul>
<li>hip</li>
<li>llvmPackages_rocm.clang</li>
<li>llvmPackages_rocm.clang-unwrapped</li>
<li>llvmPackages_rocm.clangNoCompilerRt</li>
<li>llvmPackages_rocm.compiler-rt</li>
<li>llvmPackages_rocm.lld</li>
<li>llvmPackages_rocm.llvm</li>
<li>rocclr</li>
<li>rocm-comgr</li>
<li>rocm-device-libs</li>
<li>rocm-opencl-icd</li>
<li>rocm-opencl-runtime</li>
<li>rocm-runtime</li>
<li>rocminfo</li>
</ul>
</details>

**Pull request merged**